src/models/users

```diff
@@ -0,0 +1,217 @@
+    userId: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+        required: true
+    },
+    rate: {
+        type: number,
+        enum: [1, 2, 3, 4, 5],
+        required: true
+    }
+})
+
+const baseUser = mongoose.Schema({
+    email: {
+        type: String,
+        unique: true,
+        required: true
+    },
+    hashedPassword: {
+        type: String,
+        required: true
+    },
+    provider: {
+        type: String,
+        enum: ['email', 'google', 'twitter'],
+        default: 'email',
+        required: true
+    },
+    providerId: {
+        type: String,
+    },
+    isVerified: {
+        type: Boolean,
+        required: true
+    },
+    createdEvents: [{
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "Event",
+
+    }],
+    followers: [{
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+    }],
+    createdFunds: [{
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User"
+    }]
+
+}, { toJSON: { virtuals: true } }, { toObject: { virtuals: true } });
+
+const BaseUser = mongoose.model("User", baseUser);
+
+
+const user = mongoose.Schema({
+
+    firstName: {
+        type: string,
+        required: true
+    },
+    lastName: {
+        type: string,
+        required: true
+    },
+    // fullname: virtual property
+
+    profileImage: {
+        type: string,
+    },
+    dateOfBirth: {
+        type: string,
+        required: true
+    },
+    preferredCities: [{
+        type: string,
+        enum: ['Adana', 'Kocaeli', 'Adiyaman', 'Konya'
+            , 'Afyonkarahisar', 'Kutahya'
+            , 'Agri', 'Malatya'
+            , 'Amasya', 'Manisa'
+            , 'Ankara', 'Kahramanmaras'
+            , 'Antalya', 'Mardin'
+            , 'Artviin', 'Mugla'
+            , 'Aydin', 'Mus'
+            , 'Balikesir', 'Nevsehir'
+            , 'Bilecik', 'Nigde'
+            , 'Bingol', 'Ordu'
+            , 'Bitlis', 'Rize'
+            , 'Bolu', 'Sakarya'
+            , 'Burdur', 'Samsun'
+            , 'Bursa', 'Siirt'
+            , 'Canakkale', 'Sinop'
+            , 'Cankiri', 'Sivas'
+            , 'Corum', 'Tekirdag'
+            , 'Denizli', 'Tokat'
+            , 'Diyarbakir', 'Trabzon'
+            , 'Edirne', 'Tunceli'
+            , 'Elazig', 'Sanliurfa'
+            , 'Erzincan', 'Usak'
+            , 'Erzurum', 'Van'
+            , 'Eskisehir', 'Yozgat'
+            , 'Gaziantep', 'Zonguldak'
+            , 'Giresun', 'Aksaray'
+            , 'Gumushane', 'Bayburt'
+            , 'Hakkari', 'Karaman'
+            , 'Hatay', 'Kirikkale'
+            , 'Isparta', 'Batman'
+            , 'Mersin', 'Sirnak'
+            , 'Istanbul', 'Bartin'
+            , 'Izmir', 'Ardahan'
+            , 'Kars', 'Igdir'
+            , 'Kastamonu', 'Yalova'
+            , 'Kayseri', 'Karabuk'
+            , 'Kirklareli', 'Kilis'
+            , 'Kirsehir', 'Osmaniye'
+            , 'Duzce'
+        ],
+        interests: [{
```

Comment:
interests and all the following properties should be defined outside of `preferredCities` property

Reply:
fix this creating cities variable